src/PHPHtmlParser/Selector/Selector.php

```diff
@@ -132,9 +132,7 @@ protected function seek(array $nodes, array $rule, array $options): array
         /** @var InnerNode $node */
         foreach ($nodes as $node) {
             // check if we are a leaf
-            if ($node instanceof LeafNode ||
-                ! $node->hasChildren()
-            ) {
+            if (! $node->hasChildren()) {
```

Comment:
LeafNodes do not have the hasChildren method. We can not remove that instance of check

Reply:
OK. My bad :)